packages/wdio-browserstack-service/src/service.ts

```diff
@@ -111,64 +144,113 @@ export default class BrowserstackService implements Services.ServiceInstance {
         // added to maintain backward compatibility with webdriverIO v5
         this._browser = browser ? browser : globalThis.browser
 
-        const readTcgAuthConfigToGlobal = () => {
+        // Support for browser.ai:
+        this._browser.addCommand('ai', async (userInput: string) => {
 
-            const authResult = JSON.parse(process.env.TCG_AUTH_RESULT || '{}')
-            if (Object.keys(authResult).length === 0) {
-                BStackLogger.debug('TCG Auth result is empty')
-                return { isAuthenticated: false, isHealingEnabled: false }
+            if (!(SUPPORTED_BROWSERS_FOR_AI.includes((caps as any).browserName))) {
+                BStackLogger.warn('Browserstack AI is not supported for this browser')
+                return
             }
-            return authResult
-        }
-
-        const authInfo = readTcgAuthConfigToGlobal()
-        const isAuthenticated = authInfo.isAuthenticated
-        const isHealingEnabled = authInfo.isHealingEnabled
-        if (isAuthenticated && isHealingEnabled) {
-            const sessionToken = authInfo.sessionToken
-            await aiSDK.BrowserstackHealing.setToken(this._browser.sessionId, sessionToken, TCG_URL)
-
-            const browser:any = this._browser as WebdriverIO.Browser
-            browser.overwriteCommand('findElement', async (orginalFunc: (arg0: string, arg1: string) => any, using: string, value: string) => {
-                const sessionId = browser.sessionId
-                const tcgInfo = {
-                    tcgRegion: 'use',
-                    tcgUrl: TCG_URL,
+
+            aiSDK.AISDK.configure({
+                domain: 'https://tcg-preprod.bsstag.com',
+                platform: 'desktop',
```

Comment:
How do we pass mobile/desktop values here?

Reply:
This has been fixed using one of the existing functions that checks the platform type to differentiate between the Automate and App Automate sessions.